### DISCUSSION TITLE
Question: Coming from homeassistant-solax-modbus

### DISCUSSION THREAD

**Discussion by wawyed** [Pho3niX90/solis_modbus#42] [Q&A]
Is there any benefit for me to use this integration if I'm using the https://github.com/wills106/homeassistant-solax-modbus already?

Just a genuine question to see if it's worth doing so.

**Comment by 0rangutan** [comment]
Yes, if you have a Solis inverter - you may prefer to run this instead.
This integration is Solis focussed and has a better implementation of charge/discharge times, plus now syncs the time on your inverter if it drifts.
You can run both for a while and see which you prefer.

**Comment by wawyed** [comment]
Hmm not sureI can run both at the same time? Surely only one can access the modbus at the same time?

**Comment by 0rangutan** [comment]
It will work fine for read access.  Just one TCP/IP connection is supported by the data logger but as both integrations come from the single HA instance IP it isn't a problem.

**Comment by kiloptero** [comment]
its is possible to use two dataloggers? one for solis cloud and other for
solis modbus?

dont trust solis api

Andres Roepke Del Solar.
Ingeniero Civil Industrial
Ingeniero Electronico
Mobile 88196932

**Comment by 0rangutan** [comment]
Not directly.
Instead, use one data logger and split it using Waveshare devices:
https://github.com/alienatedsec/solis-ha-modbus-cloud
Then one talks to Solos Cloud and the other to HA.
This is my setup following the above:
https://github.com/alienatedsec/solis-ha-modbus-cloud/discussions/24

**Comment by kiloptero** [comment]
im not able to create that... but change the datalogger will be easy
i already have 2 s2 and s3 wifi
the idea is to use s2 lan for ha and s3 for solid cloud

Andres Roepke Del Solar.
Ingeniero Civil Industrial
Ingeniero Electronico
Mobile 88196932

**Comment by Pho3niX90** [comment]
> Is there any benefit for me to use this integration if I'm using the https://github.com/wills106/homeassistant-solax-modbus already?
> 
> Just a genuine question to see if it's worth doing so.

To shortly answer your question. 

It comes down to what you are after. This was created to have fast data, so that automations aren't delayed, it's specifically focused only on solis inverters. 

I am obviously biased, but wills was the first one I used before writing this integration.

**Comment by Pho3niX90** [comment]
> im not able to create that... but change the datalogger will be easy
> i already have 2 s2 and s3 wifi
> the idea is to use s2 lan for ha and s3 for solid cloud
> 
> Andres Roepke Del Solar.
> Ingeniero Civil Industrial
> Ingeniero Electronico
> Mobile 88196932
> 
> 
> On Sat, Mar 16, 2024 at 16:57 0rangutan ***@***.***> wrote:
> 
> > Not directly.
> > Instead, use one data logger and split it using Waveshare devices:
> > https://github.com/alienatedsec/solis-ha-modbus-cloud
> > Then one talks to Solos Cloud and the other to HA.
> > This is my setup following the above:
> > alienatedsec/solis-ha-modbus-cloud#24
> > <https://github.com/alienatedsec/solis-ha-modbus-cloud/discussions/24>
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/Pho3niX90/solis_modbus/issues/41#issuecomment-2002108447>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/AIOTKWXIK7JBRBNCVLZL5S3YYSP37AVCNFSM6AAAAABEZMA23KVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBSGEYDQNBUG4>
> > .
> > You are receiving this because you commented.Message ID:
> > ***@***.***>
> >
> 

Haven't attempted this myself, but since you already have the hardware, you can definitely try it.

**Comment by wawyed** [comment]
> > Is there any benefit for me to use this integration if I'm using the https://github.com/wills106/homeassistant-solax-modbus already?
> > Just a genuine question to see if it's worth doing so.
> 
> To shortly answer your question.
> 
> It comes down to what you are after. This was created to have fast data, so that automations aren't delayed, it's specifically focused only on solis inverters.
> 
> I am obviously biased, but wills was the first one I used before writing this integration.

What was missing from the other one that made you create this one?

**Comment by Pho3niX90** [comment]
I cant recall, but was mostly performance and granularity of the data related, but you can see the current registers here -> https://solis-modbus.readthedocs.io/

And as 0rangutan mentioned, it now syncs the inverter time as well when it drifts.
